rust/src/pgsql/pgsql.rs

```diff
@@ -164,6 +173,17 @@ impl PgsqlState {
             backend_pid: 0,
             state_progress: PgsqlStateProgress::IdleState,
             tx_index_completed: 0,
+            events: 0,
+        }
+    }
+
+    /// Set an event on the most recent transaction
+    fn set_event(&mut self, event: PgsqlEvent) {
+        if let Some(tx) = self.transactions.back_mut() {
+            tx.tx_data.set_event(event as u8);
+            self.events += 1;
+        } else {
+            SCLogDebug!("PGSQL: trying to set event {} on non-existing transaction", event as u8);
```

Comment:
I saw that most protos only add events to existing transactions. What if there's something with the very first request - which would lead to the creation of the transaction, that happens in case of successful parsing for pgsql, would it make sense to save those somewhere?

Reply:
I may create a transaction for the sole purpose of having the event...
Not sure it is the best way...

Reply:
This is about the only way right now. Frames will help at some point when events are frames get logged:

https://redmine.openinfosecfoundation.org/issues/5826